### PR TITLE
Docker Build CircleCI Script

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
Added CircleCI script that builds and pushes v2xhub docker images with tags based on branch name.

## Related Issue

#174 
## Motivation and Context

Previously the CircleCI script only compiled the v2xhub code partially. The new script using the dockerfile to completely build and push the docker image for v2xhub which allows us to instantly diagnose docker build failures and allows for easier testing in tagged remote images for each git branch. 
## How Has This Been Tested?
Commit and ran CircleCI script and successfully pushed docker image to docker hub.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml
Comment (paulbourelly999) [inline .circleci/config.yml:119]: Initially I thought everything I was removing from the CircleCI script was duplicated somewhere in the dockerfile but it looks like a couple things like this and the Run Sonar Scanner step may be unique to the CircleCI script so I will investigate their purpose and see if I need to keep them. I will also investigate the sonar-scanner checkout and build in the dockerfile to see if that is necessary since we don't run the sonar-scanner in the dockerfile

File: .circleci/config.yml
Comment (paulbourelly999) [inline .circleci/config.yml:121]: Might be necessary to keep this step since it is not replicated in the dockerfile. Also will investigate whether the V2XHub code needs to be compiled to be scanned by the sonar scanner tool.

File: .circleci/config.yml
Comment (paulbourelly999) [reply .circleci/config.yml]: Added the old script back as well and will write an issue to investigate Sonar-scanner tool separately for Dockerfile and CircleCI script.

File: .circleci/config.yml
Comment (paulbourelly999) [reply .circleci/config.yml]: Added the old script back as well and will write an issue to investigate Sonar-scanner tool separately for Dockerfile and CircleCI script

Comment (dan-du-car) [reply .circleci/config.yml]: @paulbourelly999  I do not think we need a docker push to  push the image to dockerhub at the circleCI config. The dockerhub itself will automatically pickup the changes from a configured github branch.

**Review by dan-du-car — APPROVED** [review]

**Review by jwillmartin — APPROVED** [review]

**Comment by paulbourelly999** [comment]
Commits are squashed and branch is ready to be merged when the last reviewer is ready to merge it. Check with @arayama  and @animeshb27 before merge because I believe they are testing the develop branch after the CARMACloudPlugin merge and may want to hold off on any other merges into develop until their testing wraps up.

**Review by SaikrishnaBairamoni — APPROVED** [review]

**Comment by SaikrishnaBairamoni** [comment]
Merged.

**Pull request merged**
